### DISCUSSION TITLE
does it support -O2 without -g?

### DISCUSSION THREAD

**Discussion by dkuo123** [jeremy-rifkin/cpptrace#75] [General]
most production executable are optimized without symbol, that's when stack trace is valuable. the dev env usually is debug build and have core file to look. can you lib hook to signal handler and print out segv call stack, with function name, filename, line#?

**Comment by jeremy-rifkin** [comment]
Hi @dkuo123,
Thanks for your question, I'll do my best to answer your two questions:
- Debug symbols are generally needed in order to get information about stack frames. Sometimes function names can be gotten from export information, but that will only be function names and no file/line number information.
- The stack traces produced by the library should be reasonably useful even under -O2 -g since it can show inlined calls.
- Cpptrace can generate stack traces from a signal handler. Signal-safe stack tracing is very tricky to do so I have written up instructions for how to do this safely at https://github.com/jeremy-rifkin/cpptrace#signal-safe-tracing and https://github.com/jeremy-rifkin/cpptrace/blob/main/signal-safe-tracing.md.
